File: tests/fixtures/retrieval/polyglot/java/Sample.java
package polyglot;

import java.util.List;
import java.util.Map;

public class Sample {
    public static double computeTotal(List<Map<String, Number>> items) {
        double total = 0.0;
        for (Map<String, Number> it : items) {
            double price = it.getOrDefault("price", 0).doubleValue();
            double qty = it.getOrDefault("qty", 1).doubleValue();
            total += price * qty;
        }
        return total;
    }
}

class OrderProcessor {
    private final double taxRate;

    public OrderProcessor(double taxRate) {
        this.taxRate = taxRate;
    }

    public double process(List<Map<String, Number>> items) {
        return Sample.computeTotal(items) * (1 + taxRate);
    }
}

File: tests/fixtures/retrieval/polyglot/rust/sample.rs
pub struct Item {
    pub price: f64,
    pub qty: u32,
}

pub fn compute_total(items: &[Item]) -> f64 {
    let mut total = 0.0;
    for it in items {
        total += it.price * it.qty as f64;
    }
    total
}

pub struct OrderProcessor {
    pub tax_rate: f64,
}

impl OrderProcessor {
    pub fn process(&self, items: &[Item]) -> f64 {
        compute_total(items) * (1.0 + self.tax_rate)
    }
}
